www-reputation-credentials-protocol.md

```diff
@@ -0,0 +1,265 @@
+- the routing & payment path construction
+- the HTLC forward phase (i.e ["Accepting and Forwarding a Payment"](https://github.com/lightning/bolts/blob/master/04-onion-routing.md#accepting-and-forwarding-a-payment))
+- the HTLC settlement phase (i.e ["Removing an HTLC"](https://github.com/lightning/bolts/blob/master/02-peer-protocol.md#removing-an-htlc-update_fulfill_htlc-update_fail_htlc-and-update_fail_malformed_htlc))
+
+## Credentials dissemination phase
+
+The dissemination phase consists of the satisfaction of an acquisition method announced by the
+routing hop by the HTLC sender.
+
+```
+			2.
+			    BOLT-12 offers
+              -------------------------------------------
+             /	   3.				         \
+	 ___V_____    HTLC     _________    HTLC     _____\____
+	|	  |---------->|	        |---------->|	       |
+	|	  |           |         |           |	       |
+	|  Alice  | 1.        |   Bob   |           |  Caroll  |
+	|	  |  gossips  |	        |  gossips  |	       |
+	|_________|<----------|_________|<----------|__________|
+	    ^					   /
+	     \					  /
+	      ------------------------------------
+		       onions(credentials)
+		    4.
+
+```
+
+In this diagram:
+1. Alice discovers Caroll's `routing_policy` gossip.
+2. Alice fetches an offer from Caroll to pay the `credential_to_liquidity_unit` announced.
+3. Alice sends a HTLC to Caroll, the onion payload contains unsigned credentials.
+4. Caroll receives the HTLC, counter-signed the credentials and send them back to Alice by onions.
+
+The [reply_path](https://github.com/lightning/bolts/pull/765) between Alice-Caroll to transfer back
+the finalized credentials could be communicated during the offer exchage or the HTLC send.
+
+In this topology, Bob is Alice's LSP and credentials to route through him is assumed to be have been
+acquired by Alice in a prelimary dissemination phase. Alternatively, Bob could confer a set of free
+credentials to Alice due to the LSP-spoke trust-enhanced relationship.
+
+## Building credentials-enhanced payment paths
+
+The introduction of credentials assigned to each routing hops require some modification in the
+payment path construction algorithms.
+
+```
+						 	 _______
+							|	|  channel_update::fee_base_msat: 10
+							|  Bob  |
+							|_______|  routing_policy::credential_to_liquidity_unit: 2
+
+	 _________  Bob: 100 credentials					 	 				 _________
+	|	  | /					 ________							|	  |
+	|	  |/					|	 |  channel_update::fee_base_msat: 10			|	  |
+	|  Alice  |---- Caroll: 30 credentials  	| Caroll |							|   Eve   |
+	|	  |\					|________|  routing_policy::credential_to_liquidity_unit: 3	|	  |
+	|_________| \													|_________|
+		    Dave: 250 credentials
+							 ________
+							|	 |  channel_update::fee_base_msat: 10
+							|  Dave  |
+							|________|  routing_policy::credential_to_liquidity_unit: 1
+
+```
+
+In this diagram, Alice would like to send a 1 BTC HTLC to Eve. Bob, Caroll, Dave are all intermediary valid
```

Comment:
nit: eve is usually evil, so i was actually seeking for the attack description somewhere :)

Reply:
Run out of my stack of English names starting by E. Eliot, Ethan, Eric, yeah...okay.